lib/Doctrine/Common/Lexer/Token.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Common\Lexer;
+
+use function in_array;
+
+/** @template T of string|int */
+final class Token
```

Comment:
Because I'm reusing this name, I need to rename the type alias in `AbstractLexer` to `ArrayToken`, which should break static analysis.